### ISSUE TITLE
SerializedObjectNotCreatableException after domain reload

### ISSUE THREAD

**Issue by hailin-lee** [TylerTemp/SaintsField#10]
After Apply SaintsEditor, each time modify come code, after domain reload, SerializedObjectNotCreatableException is logged as below:

SerializedObjectNotCreatableException: Object at index 0 is null
UnityEditor.Editor.CreateSerializedObject () (at /Users/bokken/build/output/unity/unity/Editor/Mono/Inspector/Editor.cs:698)
UnityEditor.Editor.GetSerializedObjectInternal () (at /Users/bokken/build/output/unity/unity/Editor/Mono/Inspector/Editor.cs:670)
UnityEditor.Editor.get_serializedObject () (at /Users/bokken/build/output/unity/unity/Editor/Mono/Inspector/Editor.cs:564)
SaintsField.Editor.SaintsEditor.OnEnable () (at ./Library/PackageCache/today.comes.saintsfield@4d69fe59bd/Editor/SaintsEditor.cs:129)

It seems something buggy inside SaintsEditor.OnEnable().

**Comment by TylerTemp** [comment]
Hi, thanks for reporting.

I did notice this kind of unexpected behavior sometimes, but it's kinda unstable to re-produce.

I tried:

*   Unity: 2022.2.0f
*   SaintsField 2.3.4
*   Enable `SaintsEditor` using Window - Saints menu

Attach this to an empty object in an empty scene:

```csharp
public class Anything: MonoBehaviour
{
    [PropRange(0, 10)]
    [PlayaDisableIf]
    public int value3;

    [PlayaDisableIf] public string value4;

    public void Start()
    {
        Debug.Log("Hello Again!");
    }
}
```

Wait for Unity to compile.

Then modify the function and back to unity, no error gives. modify the field, no errors either. (Tested on IMGUI & UI Toolkit)

------------------------------------

Sadly, somehow I can not re-produce it even I've seen it before (((

Appreciated if you can provide a guide of how to re-produce it, with the information:

*   Unity Version: **Your version here**
*   SaintsField Version: **Find it in Window-Package Manager**
*   Did you use IMGUI or UI Toolkit? A quick way to find out is using

```csharp
[FieldType(typeof(SpriteRenderer))]
public GameObject sr;
```

IMGUI looks like (two selector buttons at the end):

![image](https://github.com/TylerTemp/SaintsField/assets/6391063/637b6bb3-4dcd-4370-a7f2-4ae840087dbe)

UI Toolkit looks like (one selector button at the end):

![image](https://github.com/TylerTemp/SaintsField/assets/6391063/9b1f53ed-3b38-4acd-b83d-3e460be22973)

And if possible, a code fragment and a way to re-produce it:

```csharp
// Your code
```

Thanks in advance!

**Comment by TylerTemp** [comment]
Hi @hailin-lee , this BUG might get fixed in 2.3.5. Without more information I can not identify which cause this problem, so please have a try in new version.

**Comment by TylerTemp** [comment]
Closed due to lack of activity.
